zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/expression/VariableStateEvaluationContext.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.zeebe.engine.processing.expression;
+
+import io.camunda.zeebe.engine.state.immutable.VariableState;
+import io.camunda.zeebe.util.buffer.BufferUtil;
+import org.agrona.DirectBuffer;
+
+public class VariableStateEvaluationContext implements ScopedEvaluationContext {
+
+  private final VariableState variableState;
+  private long scopeKey;
+
+  public VariableStateEvaluationContext(final VariableState variableState) {
+    this.variableState = variableState;
+  }
+
+  public VariableStateEvaluationContext(final VariableState variableState, final long scopeKey) {
+    this.variableState = variableState;
+    this.scopeKey = scopeKey;
+  }
+
+  @Override
+  public DirectBuffer getVariable(final String variableName) {
+    if (scopeKey < 0) {
+      return null;
+    }
+
+    return variableState.getVariable(scopeKey, BufferUtil.wrapString(variableName));
```

Comment:
💭 Here we are using `BufferUtil` for the variable name, but get back a value that is encoded via `MsgPack`. I think I need some more insights where we use which encoding, etc. 😅